### PR TITLE
Feature/gh 469: product module: refactor public accessors into functions and rename functions

### PR DESCRIPTION
**Pull request by MitchellScottSandre**
fixes: GH-469

### CODE REVIEW

**Comment by MitchellScottSandre** [comment]
@znikola This is just my first pass. Could you point out any key areas I am missing?

**Comment by MitchellScottSandre** [comment]
@znikola fixed minor issue. Needs to get reviewed again

File: projects/storefrontapp/src/environments/environment.ts [hunk 1]
```diff
@@ -5,5 +5,5 @@
 
 export const environment = {
   production: false,
-  occBaseUrl: ''
+  occBaseUrl: 'https://dev-com-17.accdemo.b2c.ydev.hybris.com:9002/'
```
Comment (znikola) [inline projects/storefrontapp/src/environments/environment.ts]: We should not commit this file

File: projects/storefrontlib/src/lib/product/components/product-list/container/product-list.component.spec.ts [hunk 2]
```diff
@@ -30,6 +30,10 @@ import { ProductSearchService } from '@spartacus/core';
 class MockProductSearchService {
   search = createSpy();
   searchResults$ = of();
+
+  getSearchResults(): Observable<any> {
```
Comment (znikola) [inline projects/storefrontlib/src/lib/product/components/product-list/container/product-list.component.spec.ts]: Is there a type we can use here?

File: projects/storefrontlib/src/lib/cms-lib/search-box/search-box-component.service.spec.ts [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
-  searchSuggestions$: of([]),
-  auxSearchResults$: of([]),
-  searchAuxiliary: createSpy().and.returnValue(of([])),
-  getSuggestions: createSpy().and.returnValue(of({}))
-};
+class MockProductSearchService {
+  searchAuxiliary = createSpy().and.returnValue(of([]));
+  getSuggestions = createSpy().and.returnValue(of({}));
+
+  getSearchSuggestions(): Observable<any> {
```
Comment (znikola) [inline projects/storefrontlib/src/lib/cms-lib/search-box/search-box-component.service.spec.ts]: Is there a type that we can use here?

File: projects/storefrontlib/src/lib/cms-lib/search-box/search-box-component.service.spec.ts [hunk 4]
```diff
@@ -1,19 +1,24 @@
...
+    return of();
+  }
+  getAuxSearchResults(): Observable<any> {
```
Comment (znikola) [inline projects/storefrontlib/src/lib/cms-lib/search-box/search-box-component.service.spec.ts]: Is there a type that we can use here?

File: projects/core/src/product/facade/product-review.service.ts [hunk 5]
```diff
@@ -21,7 +21,7 @@ export class ProductReviewService {
     );
   }
 
-  add(productCode: string, review: Review) {
+  add(productCode: string, review: Review): void {
```
Comment (znikola) [inline projects/core/src/product/facade/product-review.service.ts]: Nice

File: projects/storefrontapp/src/environments/environment.ts (hunk 1 above)
Comment (MitchellScottSandre) [reply projects/storefrontapp/src/environments/environment.ts]: Didn't mean to, thanks

**Comment by MitchellScottSandre** [comment]
@znikola made changes you requested

File: projects/core/src/product/facade/product-search.service.spec.ts [hunk 6]
```diff
@@ -73,13 +73,13 @@ describe('ProductSearchService', () => {
   ));
 
   it('should be able to get search results', () => {
-    service.searchResults$.subscribe(results => {
+    service.getSearchResults().subscribe(results => {
```
Comment (znikola) [inline projects/core/src/product/facade/product-search.service.spec.ts]: It should be like this:
```typescript
let result: type?;
 service.getSearchResults().subscribe(results => result = results).unsubscribe()
      expect(results).toEqual(mockSearchResults);
      expect(results).toEqual(mockSearchResults);
```

The point is that this test can silently pass if the nothing is returned from the subscribed observable.

File: projects/core/src/product/facade/product-search.service.spec.ts [hunk 7]
```diff
@@ -73,13 +73,13 @@ describe('ProductSearchService', () => {
...
       expect(results).toEqual(mockSearchResults);
     });
   });
 
   it('should be able to get auxiliary search results', () => {
-    service.auxSearchResults$.subscribe(results => {
+    service.getAuxSearchResults().subscribe(results => {
```
Comment (znikola) [inline projects/core/src/product/facade/product-search.service.spec.ts]: The same as above

**Pull request merged**